doc/doxygen/src/driver-guide.md

```diff
@@ -0,0 +1,381 @@
+Writing a Device Driver in RIOT                                 {#driver-guide}
+===============================
+
+This document describes the requirement, design objectives, and some
+non-function details when writing device drivers in/for RIOT. The term device
+driver in this context includes all 'CPU-external' devices connected to the CPU
+typically via peripherals like SPI, I2C, UART, GPIO, and similar. CPU
+peripherals itself are in RIOT not considered to be device drivers, but
+peripheral or low-level drivers. Typical devices are network devices like
+radios, Ethernet adapters, sensors, and actuators.
+
+[TOC]
+
+# General Design Objectives                   {#driver-guide-design-objectives}
+
+Device drivers should be as easy to use as possible. This implies an
+'initialize->ready' paradigm, meaning, that device drivers should be ready to use
+and in operation right after they have been initialized. On top, devices should
+work with physical values wherever possible. So e.g. sensors should return
+already converted values in some physical unit instead of RAW data, so that
+users can work directly with data from different devices directly without having
+to care about device specific conversion.
+
```

Comment:
```suggestion

However, please avid the use of `float` or `double`. Instead, multiply to the next SI unit. E.g. if an ADC would return values like "1.23 V", chose to return "1230 mV" instead.
```

Reply:
Oh sorry this was just about moving the file without changing it's contents, right?
If so, please ignore (We should still add this imho)

Reply:
+1 for both adding it, and for adding it as separate PR ;-)